ts/src/pro/krakenfutures.ts

```diff
@@ -0,0 +1,1281 @@
+            'cost': undefined,
+            'fee': {
+                'rate': undefined,
+                'cost': this.safeString (trade, 'tradeFee'),
+                'currency': this.safeString (trade, 'feeCurrency'),
+            },
+        }, market);
+    }
+
+    handleOrder (client: Client, message) {
+        //
+        //  update (verbose)
+        //
+        //    {
+        //        "feed": "open_orders_verbose",
+        //        "order": {
+        //            "instrument": "PI_XBTUSD",
+        //            "time": 1567597581495,
+        //            "last_update_time": 1567597581495,
+        //            "qty": 102.0,
+        //            "filled": 0.0,
+        //            "limit_price": 10601.0,
+        //            "stop_price": 0.0,
+        //            "type": "limit",
+        //            "order_id": "fa9806c9-cba9-4661-9f31-8c5fd045a95d",
+        //            "direction": 0,
+        //            "reduce_only": false
+        //        },
+        //        "is_cancel": true,
+        //        "reason": "post_order_failed_because_it_would_be_filled"
+        //    }
+        //
+        // update
+        //
+        //    {
+        //        "feed": "open_orders",
+        //        "order": {
+        //          "instrument": "PI_XBTUSD",
+        //          "time": 1567702877410,
+        //          "last_update_time": 1567702877410,
+        //          "qty": 304.0,
+        //          "filled": 0.0,
+        //          "limit_price": 10640.0,
+        //          "stop_price": 0.0,
+        //          "type": "limit",
+        //          "order_id": "59302619-41d2-4f0b-941f-7e7914760ad3",
+        //          "direction": 1,
+        //          "reduce_only": true
+        //        },
+        //        "is_cancel": false,
+        //        "reason": "new_placed_order_by_user"
+        //    }
+        //
+        let orders = this.orders;
+        if (orders === undefined) {
+            const limit = this.safeInteger (this.options, 'ordersLimit');
+            orders = new ArrayCacheBySymbolById (limit);
+            this.orders = orders;
+        }
+        const order = this.safeValue (message, 'order');
+        const marketId = this.safeStringLower (order, 'instrument');
+        if (marketId !== undefined) {
+            const messageHash = 'open_orders';
+            const symbol = this.safeSymbol (marketId);
+            const orderId = this.safeString (order, 'order_id');
+            const previousOrders = this.safeValue (orders.hashmap, symbol, {});
+            const previousOrder = this.safeValue (previousOrders, orderId);
+            if (previousOrder === undefined) {
+                const parsed = this.parseWsOrder (order);
+                orders.push (parsed);
+                client.resolve (orders, messageHash);
+            } else {
+                const trade = this.parseWsTrade (order);
```

Comment:
can we move this else branch to a method called something like `updateCachedOrder`? The handle method should be kept simple for clarity's sake

Reply:
yeah, i was going to make a unified method for it, I wanted to make sure that it was something that we should have in there before I did that though